Add/link control

gigitux opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

gigitux committed:
Page view: Add link field control

andrewserong committed:
Drag and Drop: When dragging a mix of video, audio, and image blocks, create individual blocks as appropriate (#65144)

* Drag and Drop: When dragging a mix of video, audio, and image blocks, create individual blocks as appropriate

* Consolidate into the fallback transform for the File block

Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: kevin940726 <kevin940726@git.wordpress.org>
Co-authored-by: swissspidy <swissspidy@git.wordpress.org>

talldan committed:
Disallow setting grid block rows/columns to zero (#65217)

* Disallow setting grid block rows/columns to zero

* Prevent issues with grid block column/row count by ensuring a proper minimum

Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>

vipul0425 committed:
Image: Adds the block controls for uploading image. (#64320)

* feat: Adds the block controls for uploading image.

* feat: Adds the size based placeholder logic.

* fix: The layout design for small images and adds grid support.

* refactor: Remove resize observer and add lockURLControls support.

* refactor: Make use of the existing MediaReplaceFlow component.

* fix: The mediaPlaceholder location after merge.

* fix: The initial flash in small size.

* feat: Removes the redundant css and redefine the placeholder logic.

* fix: Adressed the PR feedbacks.

* fix: Removes the effective width logic and moves observer inside of placeholder.

* fix: The e2e test failures

* Update packages/block-library/src/image/edit.js

Co-authored-by: Aki Hamano <54422211+t-hamano@users.noreply.github.com>

---------

Co-authored-by: vipul0425 <vipulgupta003@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>
Co-authored-by: annezazu <annezazu@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

kevin940726 committed:
Fix resizing to max width in classic themes (#64819)

co-authored-by: kevin940726 <kevin940726@git.wordpress.org>Co-authored-by: aaronrobertshaw <aaronrobertshaw@git.wordpress.org>Co-authored-by: andrewserong <andrewserong@git.wordpress.org>Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
* Fix resizing to max width in classic themes

* Fix padding and max-width

* Reverse the condition

* Try ignoring if layout doesn't exist

t-hamano committed:
Fix image block crash (#65222)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: kevin940726 <kevin940726@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>

gigitux committed:
Merge branch 'add/@wordpress-fields' of github.com:gigitux/gutenberg into add/link-control

ciampo committed:
DatePicker: better hover/focus styles (#65117)

* DatePicker: better hover/focus styles

* CHANGELOG

* Use after pseudo-element to avoid conflict with calendar event dot

* Make event dots compatible with high contrast mode

* Apply box sizing utility

* Use themed colors

* Use translate to prevent event dot from jumping

---

Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

ciampo committed:
Navigator Screen: warn if path doesn't follow a URL-like scheme (#65231)

* Navigator Screen: warn if path doesn't follow the expected scheme

* Add test

* CHANGELOG

---

Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>

SantosGuillamot committed:
Populate block context with inherited post type from template slug (#65062)

* Inherit post type from template slug

* Move map outside of `useSelect`

Co-authored-by: SantosGuillamot <santosguillamot@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: artemiomorales <artemiosans@git.wordpress.org>
Co-authored-by: cbravobernal <cbravobernal@git.wordpress.org>

AKSHAT2802 committed:
Remove Warning and add notice for Navigation (#63921)


Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: afercia <afercia@git.wordpress.org>

sirreal committed:
Build: Prepare build for more script modules (#65064)

- Rename the "interactivity" webpack build to "script modules".
- Output script-modules builds to `build-module` folder (adjacent to the `build` folder currently used for scripts).
- Add `wpScriptModulesExports` package.json field to packages with script modules and use it for script module builds. This follows the same basic syntax as [package.json `exports` fields](https://nodejs.org/api/packages.html#exports). Multiple module entrypoints can be exposed per package. However, it remains a custom field, so it is clear that these entrypoints are not intended for general consumption. In the future, module-only packages (interactivity, interactivity-router) can switch to using exports directly and likely add `type: module`.
- There are some difficulties with webpack recognizing `wpScriptModulesExports` directly, so packages are inspected programmatically in order to generate webpack script modules entrypoints.
- Adjust script module registration accordingly to find the generated script modules.

---

Co-authored-by: sirreal <jonsurrell@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

gigitux committed:
Merge branch 'add/@wordpress-fields' of github.com:gigitux/gutenberg into add/link-control

gigitux committed:
move slug field control to @wordpress/fields package

youknowriad commented on packages/edit-site/src/components/post-fields/index.js:
```diff
@@ -369,6 +370,13 @@ function usePostFields( viewType ) {
+				id: 'slug',
+				getValue: ( { item } ) => item.slug,
+				render: SlugView,
+				Edit: SlugEdit,
```
Any reason for exporting SlugView and SlugEdit separately, instead of exporting the whole field from the other package?

Ultimately though (later), I feel like we might not need to export the fields directly we could just have a selector to fetch the fields per post type (like we do for actions)

gigitux replied:
I see your point and I agree with you! For now, I would orient to follow [this approach](https://github.com/WordPress/gutenberg/blob/9bfd08fff59ff8c2526eac2930239f1dc87a8d4c/packages/editor/src/dataviews/fields/index.ts/#L13-L27).

> Ultimately though (later), I feel like we might not need to export the fields directly we could just have a selector to fetch the fields per post type (like we do for actions)

It makes sense!

youknowriad replied:
> I see your point and I agree with you! For now, I would orient to follow [this approach](https://github.com/WordPress/gutenberg/blob/9bfd08fff59ff8c2526eac2930239f1dc87a8d4c/packages/editor/src/dataviews/fields/index.ts/#L13-L27).

👍

gigitux committed:
Add @wordpress/fields package (#65230)

Co-authored-by: gigitux <gigitux@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>

DaniGuardiola committed:
Improve Tabs indicator animation and related utils (#64926)

* Refactor utils and switch Tabs indicator animation to `transform`.

* docs tweak

* Update packages/components/src/tabs/styles.ts

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

* Add RTL support.

* Addressed @ciampo's comments.

* Correct for antialiasing rounding.

* Make antialiasing adjustment optional.

* Use larger base value and revert antialiasing adjustment code.

* DRY RTL

* Remove RTL story (redundant since Storybook has a dynamic setting to test RTL).

* Fix bug.

* Fix bug (for real this time).

* Add changelog entry.

* De-cleverfy code.

* Sync useResizeObserver with #64943 and make useTrackElementOffsetRect resilient.

* Deduplicate utility and clean up.

* DRY antialiasing factor.

* Changelogs.

---------

Co-authored-by: DaniGuardiola <daniguardiola@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

jeryj committed:
Fix Content Only Toolbar icon focus style (#64940)

Co-authored-by: jeryj <jeryj@git.wordpress.org>
Co-authored-by: getdave <get_dave@git.wordpress.org>

mirka committed:
Remove colons from control labels (#65205)

* Remove colons from control labels in docs

* Remove label colon from Archives block

* Remove label colon in Latest Posts block

* Fix label in Legacy Widget block

* Shorten to "Legacy widget" and update test

Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: DaniGuardiola <daniguardiola@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

jorgefilipecosta committed:
Fix: Moving the last page item to the the trash causes a crash. (#65236)

co-authored-by: jorgefilipecosta <jorgefilipecosta@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

mirka committed:
Normalize block inspector controls spacing (#64526)

* Try block inspector controls spacing normalization

* Decrease specificity

* Add changelog

* Grid block: Fix when experimental grid interactivity

* Remove adjustment for bottom padding

* Revert "Add changelog"

This reverts commit 7939750f87b2d35d3f54deab4c8c91df5d49e2c8.

* Remove unnecessary margin on Layout hook toggle (#65133)

Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

gutenbergplugin committed:
Bump plugin version to 19.2.0

gutenbergplugin committed:
Update Changelog for 19.2.0

hbhalodia committed:
Add the spacing for horizontal and vertical for block gap on buttons block (#64971)

Co-authored-by: hbhalodia <hbhalodia@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: Nyiriland <nyiriland@git.wordpress.org>

Mamaduka committed:
Plugin: Remove 'function_exists' checks for methods with 'gutenberg' prefix (#65260)

* Plugin: Remove 'function_exists' checks for methods with 'gutenberg' prefix
* Move code into the correct compat file

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>

mikeybinns committed:
Scripts: Update stylelint dependency and the default config (#64828)

* update stylelint deps to latest and convert deprecated rules to @stylistic

* prevent extra rules being added

* fix rule name change

* fix scss overwrite bug

* fix stylelint disables/enables

* migrate stylistic styles to separate configs to make it easy to opt out for consumers

* fix: switch stylistic config dep for plugin

* use require.resolve for stylelint recommended config

* fix stylelint unit tests

* fix global mocks for node env

* use correct configs for tests

* add type checking for configs

* convert tests to esm

* remove unnecessary await

* fix stylelint tests

* remove experimental-vm-modules flag

* revert jest globals changes

* move utils to folder

* update readme

* update changelogs

---------

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

Unlinked contributors: 75th, vicobot-0815.

Co-authored-by: mikeybinns <mikeybinns@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>
Co-authored-by: jacobcassidy <jacobcassidy@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: paulschreiber <paulschreiber@git.wordpress.org>
Co-authored-by: stein2nd <stein2nd@git.wordpress.org>

getdave committed:
Show top level sections in List View in Zoom Out mode (#65202)

* Show top level sections in List View

* Recompute enabled clientIds when editorMode changes

* Revert initial changes

* Avoid special conditional for Zoom Out mode

AKSHAT2802 committed:
Add __next40pxDefaultSize for files in editor 4 (#65140)

* Add __next40pxDefaultSize in button for editor 3 files

* address feedback

Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>

getdave committed:
Close inserter on exiting Zoom Out to edit (#65194)

* Use experimental setting in single call location

* Close inserter on double click to exit Zoom Out

* Avoid additional store subscription

oandregal committed:
Make `wordpress/fields` a private package (#65269)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>

sirreal committed:
Build Plugin: Simplify and improve zip contents (#65232)

Include all of /build directory. Behavior is largely unchanged but logic is simplified.
Exclude directory entries. Directory structure is preserved but irrelevant directory attributes are excluded.

---
Co-authored-by: sirreal <jonsurrell@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>

gigitux committed:
Merge branch 'trunk' of github.com:WordPress/gutenberg into add/link-control

sirreal committed:
A11y: Add Script Module (#65101)

Add a `@wordpress/a11y` WordPress Script Module. The script module has the same public API as the `wp-a11y` script.

- The `domReady` package should not be necessary. Since modules are deferred, setup can be called directly when the module is evaluated.
- There is no i18n script module at this time. The necessary string "Notifications" is translated on the server and passed to the package via script module data.
- Most of the changes here are moving some functions around so that the script and the module form of this package rely on the same underlying implementations.

---

Co-authored-by: sirreal <jonsurrell@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>

t-hamano committed:
Build Tooling: Correctly generate PHP files for server-side rendering of blocks on Windows OS (#65248)

* Build Tooling: Correctly generate PHP files for server-side rendering of blocks on Windows OS

* Don't use  prefix

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: sirreal <jonsurrell@git.wordpress.org>

t-hamano committed:
Form Input: Don't use `flex-direction: row-reverse` for checkbox field (#64232)

* Form Input: Don't use `flex-direction: row-reverse` for checkbox field

* Don't create v2 variable

* Fix fixtures

* Update fixtures

* Split fixtures

* Integrate fixtures

* Remove unneccesary fixtures

* Add base fixtures for text-like inputs

All variations of the simple texts input field are added here to match the approach taken in the deprecation fixtures.

* Add base fixtures for checkbox, radio, and textarea inputs

* Add fallback style for old markup

---------

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: aaronrobertshaw <aaronrobertshaw@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: aristath <aristath@git.wordpress.org>
Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>

sirreal committed:
Remove babel from script-modules build (#65279)

Co-authored-by: sirreal <jonsurrell@git.wordpress.org>
Co-authored-by: cbravobernal <cbravobernal@git.wordpress.org>

jsnajdr committed:
Switch from UglifyJS to Terser to build the polyfill script (#65278)

Co-authored-by: jsnajdr <jsnajdr@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: sgomes <sergiomdgomes@git.wordpress.org>

greenworld committed:
Fix typo in Slot Fills documentation (#65275)


Unlinked contributors: greenworld.

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

AKSHAT2802 committed:
Add props for buttons in editor 2 (#65083)

* Add __next40pxDefaultSize for editor 2 files

* Update packages/editor/src/components/post-publish-panel/index.js

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

* update test snapshots

---------

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

jorgefilipecosta committed:
Update: Remove unused css selectors. (#65276)

co-authored-by: jorgefilipecosta <jorgefilipecosta@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

getdave committed:
Add Zoom Out toggle to editor header when experiment enabled (#65183)

* Add toggle when experiment active

* Add new square icon

* Use new icon

* Remove zoom out desktop 50% from device preview dropdown

* Fix console error for stroke-width to strokeWidth

---------

Co-authored-by: Jerry Jones <jones.jeremydavid@gmail.com>
Co-authored-by: getdave <get_dave@git.wordpress.org>
Co-authored-by: jeryj <jeryj@git.wordpress.org>
Co-authored-by: scruffian <scruffian@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>

cbravobernal committed:
Block Bindings: Try gap 0 on attribute items. (#65277)

* Try gap 0

* Use vstack property

Co-authored-by: cbravobernal <cbravobernal@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

gigitux committed:
[Fields] Migrate store and actions from editor package to fields package (#65261)

Co-authored-by: gigitux <gigitux@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>

jameskoster committed:
Align popover alt variant styling with block toolbar (#65263)

Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

t-hamano committed:
BoxControl: Fix critical error when null value is passed (#65287)

* BoxControl: Fix critical error when null value is passed

* Add Changelog entry

* Update packages/components/src/box-control/utils.ts

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

---------

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

AKSHAT2802 committed:
Add props for buttons in editor 1 (#65068)

* Add props for buttons in editor 1

* Address feedback

* remove extra class name and extra styles

Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

rohitmathur-7 committed:
Grid: In RTL languages, the resize handles point in the opposite direction  (#64995)

* Fixed the RTL resizing issue

* Removed comment, fix spacing

* Added else condition

* Revert "Fixed the RTL resizing issue"

This reverts commit 39cfde5052c71712e4d0f6337f39c523e3830785.

* Updated left and right values for justification

Co-authored-by: rohitmathur-7 <rohitmathur7@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>

madhusudhand committed:
Spacing control: Replace sides dropdwon with link button (#65193)

* replace sides dropdown with link button for spacing control

* remove sides dropdown

* lowercase the link button tooltip

* fix unit tests

mikeybinns committed:
Styleling config: Fix stylelint config missing files for npm (#65313)

* Add stylistic configs to packaged files

---------

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

Co-authored-by: mikeybinns <mikeybinns@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>

cbravobernal committed:
Patterns: Change deprecated social icons for standard in e2e (#65312)

* Change deprecated social icons for standard

* Update more tests

* Update test/e2e/specs/editor/various/adding-patterns.spec.js

Co-authored-by: Kai Hao <kevin830726@gmail.com>

* One more spec

* Update snapshots

* Revert changes to post-editor.spec.js

---------

Co-authored-by: cbravobernal <cbravobernal@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: kevin940726 <kevin940726@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

oandregal committed:
DataViews: improve UX of bundled views for Pages (#65295)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: Souptik2001 <souptik@git.wordpress.org>
Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: SaxonF <saxonafletcher@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

stokesman committed:
Split content view with meta boxes (#64351)

* Use iframe even with metaboxes

* Split content view when metaboxes are open

* Fix styling issues

Co-authored-by: t-hamano <wildworks@git.wordpress.org>

* Update e2e tests

Co-authored-by: ellatrix <ellatrix@git.wordpress.org>

* Make metaboxes a details element in short viewports

* Make metabox area resizable when viewport isn’t short

* Tweak details element usage

* Add max-height when resized by user

* Hide metabox area if all metaboxes are hidden

* Persist toggle and height state in user preferences

* Wrap contents and rename things

The added wrapping element was due to Safari clipping the drop-shadow
when `overflow: auto` was on the component root.

* Make visual editor stacking context when iframed with metaboxes

* Omit meta box area when empty instead of hiding it

* Fix meta boxes content appearing in front of UI

* Use split views only when canvas is iframed

* Don’t omit main meta box area unless all locations aren’t visible

---------

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: ellatrix <ellatrix@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: torounit <toro_unit@git.wordpress.org>
Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: arnaudbroes <arnaudbroes@git.wordpress.org>

peterwilsoncc committed:
Update minimum required version in PHP. (#65301)

* Update minimum required version in PHP.

* Use a constant.

* Test plugin meta data and constant match.

* Set constant incorrectly to demonstrate failing tests.

* Use regex to extract the constant.

* Match the version in the constant and plugin meta.

Co-authored-by: peterwilsoncc <peterwilsoncc@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

t-hamano committed:
DimensionsPanel: Apply 40px default size to UI when no spacing preset is available (#65300)

* DimensionsPanel: Apply 40px default size to UI when no spacing preset is available

* Revert unnecessary change

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

ntsekouras committed:
DataViews: Use Dropdown for views config dialog (#65314)

Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>

Mamaduka committed:
Block Editor: Use static access for selector in 'useZoomOutModeExit' (#65337)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

jorgefilipecosta committed:
Add: Typography styling support to the navigation submenu block. (#65060)

co-authored-by: jorgefilipecosta <jorgefilipecosta@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: annezazu <annezazu@git.wordpress.org>
Co-authored-by: cuemarie <plari@git.wordpress.org>

t-hamano committed:
Editor: Use hooks instead of HoC in `BlockManager` (#65349)


Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

ramonjd committed:
Global Styles: refactor site background controls and move site global styles into Background group (#65304)

Expose background styles in the top level global styles navigation menu. Background is now underneath color. Refactor background image controls and their styles into a dedicated component. This is in preparation for adding colors and other background controls later.

Co-authored-by: amitraj2203 <amitraj2203@git.wordpress.org>
Co-authored-by: mtias <matveb@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>

carolinan committed:
Query loop / Post template: Enable post format filter (#64167)

Enables filtering the query loop result by post format.

To allow querying a given post format, a new parameter, `format` is added to a new compatibility class called `Gutenberg_REST_Posts_Controller_6_7`, which extends the core class `WP_REST_Posts_Controller`.

Query loop block:
- Adds a new parameter, `format`, to the `query` attribute in block.json.
- Adds a new `FormatControls` that uses a list of supported post formats inside a `FormTokenField`. This control is placed in the `Filters` panel in the block settings sidebar.
- Adds a new utility function `gutenberg_add_format_query_vars_to_query_loop_block` to ensure that the Query loop block can pass the new `format` argument correctly to `WP_Query`. This function is hooked into `query_loop_block_query_vars`.

Post Template block:
The new `format` parameter is passed from the query loop block to the post template block as part of the `query` attribute.

---------

Co-authored-by: carolinan <poena@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: dmsnell <dmsnell@git.wordpress.org>
Co-authored-by: TimothyBJacobs <timothyblynjacobs@git.wordpress.org>
Co-authored-by: nickbohle <nickbohle@git.wordpress.org>
Co-authored-by: SergeyBiryukov <sergeybiryukov@git.wordpress.org>
Co-authored-by: justintadlock <greenshady@git.wordpress.org>

jeryj committed:
Put Replace in own group to give it borders (#64849)

Co-authored-by: jeryj <jeryj@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>
Co-authored-by: MaggieCabrera <onemaggie@git.wordpress.org>

SantosGuillamot committed:
Pass the comments query paged arg to functions (#63698)

Co-authored-by: SantosGuillamot <santosguillamot@git.wordpress.org>
Co-authored-by: ockham <bernhard-reiter@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>

colorful-tones committed:
Add JSDoc block for getSectionRootClientId in block editor package (#65219)

* Add JSDoc block for getSectionRootClientId in block editor

* Apply suggestions from code review

Co-authored-by: Dave Smith <getdavemail@gmail.com>

---------

Co-authored-by: Dave Smith <getdavemail@gmail.com>

mirka committed:
Fix Tabs styling in Font Library modal (#65330)

Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: DaniGuardiola <daniguardiola@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>

hbhalodia committed:
Fix: Button: Replace remaining 40px default size violation [Edit Site 2] (#65258)

* Fix font library modal fonts to use 40px default button size

* Fix install font button to use default 40px size

* Fix the installed fonts button size to use 40px default

* Fix upload font button size to use 40px default size

* Fix add the default button 40px size for randomize color pallete button

* Fix edit site style revision button to use 40px default size

* Fix edit site shadow components buttons to use default 40px size

* Fix the style for the button height on the font library modal

* Add !important to override the button size for 40px

Co-authored-by: hbhalodia <hbhalodia@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>

louwie17 committed:
Convert EmbedPreview component to functional component (#51325)

* Convert EmbedPreview component to functional component

* Update interactive condition and move call inline versus within useEffect

Unlinked contributors: louwie17.

Co-authored-by: ntsekouras <ntsekouras@git.wordpress.org>
Co-authored-by: kevin940726 <kevin940726@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

jameskoster committed:
Update Card radius (#65053)

Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

t-hamano committed:
Docs: Update the content of the API version 3 section in the Block API Reference (#65375)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

dependabot[bot] committed:
Bump the react-native group across 1 directory with 2 updates (#65321)

Bumps the react-native group with 2 updates in the / directory: [actions/setup-java](https://github.com/actions/setup-java) and [gradle/actions](https://github.com/gradle/actions).


Updates `actions/setup-java` from 4.2.2 to 4.3.0
- [Release notes](https://github.com/actions/setup-java/releases)
- [Commits](https://github.com/actions/setup-java/compare/6a0805fcefea3d4657a47ac4c165951e33482018...2dfa2011c5b2a0f1489bf9e433881c92c1631f88)

Updates `gradle/actions` from 4.0.0 to 4.1.0
- [Release notes](https://github.com/gradle/actions/releases)
- [Commits](https://github.com/gradle/actions/compare/af1da67850ed9a4cedd57bfd976089dd991e2582...d156388eb19639ec20ade50009f3d199ce1e2808)

---
updated-dependencies:
- dependency-name: actions/setup-java
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: react-native
- dependency-name: gradle/actions
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: react-native
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

creativecoder committed:
DocumentBar: replace icon with post type label (#65170)

* Adds post type label to DocumentBar

* Removes the post type icon from the DocumentBar

* Try fixing one of the e2e tests

* Remove IconType import and add typedef

* Adds font-weight to heading and uses darker grey for constrast

* Adjusts title overflow so it dosen't apply to post type label

* Moves font weight to h1

* Try fixing more e2e tests

* Fix another e2e

* Add fix for mid screensizes when Back button is present

* Hide post type label on small screen sizes

---------

Co-authored-by: creativecoder <grantmkin@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: stokesman <presstoke@git.wordpress.org>
Co-authored-by: jeryj <jeryj@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: afercia <afercia@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>

Mamaduka committed:
Global Styles: Update REST controller override method and backport changes from Core (#65259)

Global Styles: Update REST controller override method: Update unit tests; Update class with Core's WP_REST_Global_Styles_Controller; Inherit WP_REST_Posts_Controller, make methods public et. al; Add 6.6+ post args to register post filter; Overwrite Core endpoints with $override when registering routes.
---------

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: aaronrobertshaw <aaronrobertshaw@git.wordpress.org>
Co-authored-by: spacedmonkey <spacedmonkey@git.wordpress.org>

mirka committed:
BoxControl: Add lint rule for 40px size prop usage (#65341)

* BoxControl: Add lint rule for 40px size prop usage

* Add to readme

Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>

mirka committed:
ButtonGroup: Fix story to show what the component does (#65336)

Co-authored-by: mirka <0mirka00@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>

andrewserong committed:
Image cropping: skip making an API request if there are no changes to apply (#65384)

Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: ajlende <ajlende@git.wordpress.org>
Co-authored-by: annezazu <annezazu@git.wordpress.org>

AKSHAT2802 committed:
Add __next40pxDefaultSize for files in editor 3 (#65139)

* Add __next40pxDefaultSize for files in editor 3

* Address feedback

* change font size to 13 px

* Revert classic-theme.js changes as it will be a seperate PR and revert 13px size in style as it is default

* Wrap copy button in  InputControlSuffixWrapper

Co-authored-by: AKSHAT2802 <akshat2802@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: mirka <0mirka00@git.wordpress.org>

ciampo committed:
Rename edit-post__fade-in-animation and unify keyframe definitions (#65377)

* Remove duplicate keyframe definition across packages

* Deprecate edit-post__fade-in-animation, replace with animation__fade-in

* Add animation__fade-out mixin

dhruvang21 committed:
Add __next40pxDefaultSize for edit-widgets files (#65367)

gigitux committed:
Merge branch 'trunk' of github.com:WordPress/gutenberg into add/link-control
